js/ui/popup.js

```diff
@@ -413,6 +417,23 @@ var Popup = Overlay.inherit({
             .toggleClass(POPUP_NORMAL_CLASS, !value);
     },
 
+    _shouldUseContentAutoResizableClass() {
+        if(this.option("height") !== "auto" || this.option("shading") || this.option("fullScreen") || this.option("showTitle") || this._getToolbarItems("bottom").length) {
+            return false;
+        }
+
+        return this.option("useAutoHeightWithLimits");
+    },
+
+    _toggleContentAutoResizableClass: function() {
+        this._$wrapper.toggleClass(POPUP_AUTO_RESIZABLE_CLASS, this._shouldUseContentAutoResizableClass());
+    },
+
+    _renderAutoResizableGeometry: function() {
+        this._toggleContentAutoResizableClass();
```

Comment:
Why do you need this method? Can you insert `this._toggleContentAutoResizableClass` in the top of the `renderGeometry` method?